fe/fe-core/src/main/java/com/starrocks/persist/gson/GsonUtils.java

```diff
@@ -208,6 +212,12 @@ public class GsonUtils {
             = RuntimeTypeAdapterFactory.of(SnapshotInfo.class, "clazz")
             .registerSubtype(LakeTableSnapshotInfo.class, LakeTableSnapshotInfo.class.getSimpleName());
 
+    private static final RuntimeTypeAdapterFactory<PEntryObject> P_ENTRY_OBJECT_RUNTIME_TYPE_ADAPTER_FACTORY =
```

Comment:
It will automatically recognize the real class of the JSON with an extra field `clazz`. 
Let's say we have a list of PEntryObject, which actually filled with TableEntryObject.
```
List<PEntryObject> l = Arrays.asLists(new TablePEntryObject(1, 2));
```
If you dump the list to JSON, it should be something like this.
```
[{"d": 1, "t": 2}]
```
If you load this string to a list of PEntryObject, there's no way to know the precise class to instantiate with. 
https://www.javadoc.io/doc/org.danilopianini/gson-extras/0.2.1/com/google/gson/typeadapters/RuntimeTypeAdapterFactory.html

Reply:
got it ,👍🏻